vernac/obligations.ml

```diff
@@ -20,6 +20,12 @@ open Pp
 open CErrors
 open Util
 
+(* EJGA: This should go away, no more need for fix_exn  *)
+module Stm = struct
+    let u = (fun x -> x)
+    let get_fix_exn () = u
+end
+
```

Comment:
One question is what would happen if we image that `Vernacentries.interp` was functional; indeed the `state_id` should be a parameter so feedback could be properly done, etc... By upstream I meant the caller of `Vernacentries.interp`, that is, the `Stm`.

But going back to `fix_exn`, something I don't fully understand is why exceptions have to carry the last valid `state_id` (and all the associated error handling with the call_once stuff, etc...). Shouldn't the upper layer of the `Stm` the one in charge of finding what would the valid state be? So it seems that currently, the `Stm` knows what the good state is, but chooses to forget it and embed that information into the lower layers, which will give back the information if an error occurs. Does that sound similar to the reality? I am quite behind in understanding the Stm yet, and I am aware that error handling with futures is quite tricky.